File: joe/csharp/RockPaperScissors.cs
using System;
using System.Collections.Generic;

namespace csharp
{
	class RockPaperScissors
	{
		private static Random random = new Random();
		private static string[] choices = { Choice.Rock, Choice.Paper, Choice.Scissors };
		private static IDictionary<string, string> losingChoices
		{
			get
			{
				var choices = new Dictionary<string, string>();
				choices.Add(Choice.Rock, Choice.Scissors);
				choices.Add(Choice.Scissors, Choice.Paper);
				choices.Add(Choice.Paper, Choice.Rock);
				return choices;
			}
		}

		public static void Main()
		{
			while (true)
			{
				var input = GetInput("Rock, paper, scissors? ");

				switch (input)
				{
					case Choice.Rock:
					case Choice.Scissors:
					case Choice.Paper:
						// Okay
						break;

					case ".quit":
					case "q":
					case "stop":
					case "exit":
						Console.WriteLine("Fun playing with you, shutting down...");
						Environment.Exit(0);
						break;

					default:
						Console.WriteLine("That wasn't a valid choice, please try again.");
						continue;
				}

				var botChoice = choices[random.Next(choices.Length)];
				var losingChoice = losingChoices[input];

				if (input.Equals(losingChoice)) Console.WriteLine($"You got absolutely destroyed by the bigbrain computer, as I chose {botChoice} versus your {input}. Wanna play again?");
				else if (input.Equals(botChoice)) Console.WriteLine("Aw man, it was a tie... Let's play again.");
				else Console.WriteLine($"You beat me, human. I chose {botChoice} but you chose {input}. Now let's play again.");
			}
		}

		private static string GetInput(string prompt)
		{
			Console.Write($"\u001b[92m{prompt}\u001b[0m");
			return Console.ReadLine().Trim().ToLower();
		}
	}

	public class Choice
	{
		public const string Rock = "rock";
		public const string Paper = "paper";
		public const string Scissors = "scissors";
	}
}

File: joe/java/RockPaperScissors.java
import java.util.Scanner;
import java.util.Random;
import java.util.HashMap;
import java.util.Map;

public class RockPaperScissors {
	private static final Scanner scanner = new Scanner(System.in);
	private static final Random rand = new Random();
	private static final Choice[] choices = Choice.values();
	private static final char ansiSeparator = (char) Integer.parseInt("0001b", 16);

	private static Map<Choice, Choice> losingChoices = new HashMap<>();

	static {
		losingChoices.put(Choice.ROCK, Choice.SCISSORS);
		losingChoices.put(Choice.SCISSORS, Choice.ROCK);
		losingChoices.put(Choice.PAPER, Choice.ROCK);
	}

	public static void main(String[] args) {
		while (true) {
			var userInput = input("Rock, paper, scissors? ");

			var userChoice = Choice.fromStr(userInput);
			if (userChoice == null) {
				switch (userInput) {
					case ".quit":
					case "q":
					case "stop":
					case "exit":
						System.out.println("Fun playing with you, shutting down...");
						System.exit(0);

					default:
						System.out.println("That wasn't a valid choice, please try again.");
						continue;
				}
			}

			var botChoice = choices[rand.nextInt(choices.length)];
			var losingChoice = losingChoices.get(botChoice);
			if (userChoice.equals(losingChoice)) {
				System.out.printf(
						"You got absolutely destroyed by the bigbrain computer, as I chose %s versus your %s. Wanna play again?\n",
						botChoice.getStr(), userChoice.getStr());
			} else if (userChoice.equals(botChoice)) {
				System.out.println("Aw man, it was a tie... Let's play again.");
			} else {
				System.out.printf("You beat me, human. I chose %s but you chose %s... Now let's play again.\n",
						botChoice.getStr(), userChoice.getStr());
			}
		}
	}

	private static String input(String prompt) {
		System.out.printf("%c[92m%s%c[0m", ansiSeparator, prompt, ansiSeparator);
		return scanner.nextLine().trim().toLowerCase();
	}

	private enum Choice {
		ROCK("rock"), PAPER("paper"), SCISSORS("scissors");

		private final String choiceStr;

		Choice(String choiceStr) {
			this.choiceStr = choiceStr;
		}

		public String getStr() {
			return choiceStr;
		}

		private final static Map<String, Choice> lookup = new HashMap<>();

		static {
			for (var choice : Choice.values()) {
				lookup.put(choice.getStr(), choice);
			}
		}

		public static Choice fromStr(String str) {
			return lookup.get(str);
		}
	}
}